SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -982,13 +983,28 @@ - (void)sendMessageToService:(TSOutgoingMessage *)message
         OWSAssert(deviceMessages.count > 0);
     }
 
+    if (deviceMessages.count == 0) {
+        DDLogWarn(@"%@ Sending a message with no device messages.", self.logTag);
```

Comment:
Is the only time that we expect this when sending a sync message to ourself for the first time? Or are there other occasions?

Reply:
I think this might occur in other rare edge cases, e.g. after unlinking all linked devices and after trying and failing to link a device.

Reply:
I'll leave a comment to this effect.